lib/ramble/ramble/test/end_to_end/env_var_builtin.py

```diff
@@ -0,0 +1,214 @@
+            simple_test:
+              variables:
+                n_nodes: 1
+              env-vars:
+                set:
+                  MY_VAR: 'TEST'
+  spack:
+    concretized: true
+    packages: {}
+    environments: {}
+"""
+    workspace_name = 'test_env_var_command'
+    with ramble.workspace.create(workspace_name) as ws:
+        ws.write()
+
+        config_path = os.path.join(ws.config_dir, ramble.workspace.config_file_name)
+
+        with open(config_path, 'w+') as f:
+            f.write(test_config)
+        ws._re_read()
+
+        workspace('setup', '--dry-run', global_args=['-w', workspace_name])
+
+        experiment_root = ws.experiment_dir
+        exp1_dir = os.path.join(experiment_root, 'interleved-env-vars', 'test_wl', 'simple_test')
+        exp1_script = os.path.join(exp1_dir, 'execute_experiment')
+        exp2_dir = os.path.join(experiment_root, 'interleved-env-vars', 'test_wl2', 'simple_test')
+        exp2_script = os.path.join(exp2_dir, 'execute_experiment')
+        exp3_dir = os.path.join(experiment_root, 'interleved-env-vars', 'test_wl3', 'simple_test')
+        exp3_script = os.path.join(exp3_dir, 'execute_experiment')
+
+        import re
+        export_regex = re.compile(r'export MY_VAR=TEST')
+        cmd1_regex = re.compile('bar >>')
+        cmd2_regex = re.compile('baz >>')
+        cmd3_regex = re.compile('foo >>')
+
+        # Assert experiment 1 has exports before commands
+        with open(exp1_script, 'r') as f:
+            cmd_found = False
+            export_found = False
+            for line in f.readlines():
+                if not export_found and export_regex.search(line):
+                    assert not cmd_found
+                    export_found = True
+                if export_found and cmd1_regex.search(line):
+                    cmd_found = True
+            assert cmd_found and export_found
+
+        # Assert experiment 2 has commands before exports
+        with open(exp2_script, 'r') as f:
+            cmd_found = False
+            export_found = False
+            for line in f.readlines():
+                if not cmd_found and cmd2_regex.search(line):
+                    assert not export_found
+                    cmd_found = True
+                if cmd_found and export_regex.search(line):
+                    export_found = True
+            assert cmd_found and export_found
+
+        # Assert experiment 3 has exports before commands
+        with open(exp3_script, 'r') as f:
+            cmd_found = False
+            export_found = False
+            for line in f.readlines():
+                if not export_found and export_regex.search(line):
+                    assert not cmd_found
+                    export_found = True
+                if export_found and cmd3_regex.search(line):
+                    cmd_found = True
+            assert cmd_found and export_found
+
+
+def test_configvar_dry_run(mutable_config, mutable_mock_workspace_path):
```

Comment:
This test looks replicated below in test_configvar_dry_run. Do we still need it here too?

Reply:
Oops, that was an earlier version that crept back in when I did `rebase -i`. Fixing, see next push.